File: back/src/main/java/com/back/domain/schedule/dto/ShowScheduleResponse.java
package com.back.domain.schedule.dto;

import com.back.domain.schedule.entity.Schedule;

import java.time.LocalDateTime;

public record ShowScheduleResponse(
            Long concertId,
            Long scheduleId,
            int round,
            LocalDateTime scheduleDate,
            long remainingSeats
) {
    public static ShowScheduleResponse from(Schedule schedule,long remainingSeats) {
        return new ShowScheduleResponse(
                schedule.getConcert().getConcertId(),
                schedule.getScheduleId(),
                schedule.getRound(),
                schedule.getScheduleDate(),
                remainingSeats
        );

    }
}

File: back/src/main/kotlin/com/back/domain/schedule/dto/ShowScheduleResponse.kt
package com.back.domain.schedule.dto

import com.back.domain.schedule.entity.Schedule
import java.time.LocalDateTime

data class ShowScheduleResponse(
    val concertId: Long?,
    val scheduleId: Long?,
    val round: Int,
    val scheduleDate: LocalDateTime,
    val remainingSeats: Long
) {
    companion object {
        @JvmStatic
        fun of(schedule: Schedule, remainingSeats: Long): ShowScheduleResponse =
            ShowScheduleResponse(
                concertId = schedule.concert.concertId,
                scheduleId = schedule.scheduleId,
                round = schedule.round,
                scheduleDate = schedule.scheduleDate,
                remainingSeats = remainingSeats
            )
    }
}
